walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,156 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    ArrayList<TimeRange> slots = new ArrayList<TimeRange>();
+
+    ArrayList<Event> eventsArr = sortEventsByStart(events);
+
+    if (request.getDuration() > TimeRange.WHOLE_DAY.duration()) {
+      return slots;
+    }
```

Comment:
It's generally more readable to check invalid conditions at the very top of the method and immediately return before creating any local variables.

I would also add a comment to explain what this check is and why requests for meetings longer than a day are invalid.

```suggestion
    // Request duration is longer than a day, don't return anything (why?).
    if (request.getDuration() > TimeRange.WHOLE_DAY.duration()) {
      return new ArrayList<TimeRange>();
    }
    
    ArrayList<TimeRange> slots = new ArrayList<TimeRange>();
    ArrayList<Event> eventsArr = sortEventsByStart(events);    
```